bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/firmware/ProgressCallback.java

```diff
@@ -74,4 +87,16 @@
      * Callback operation to indicate that the firmware update was canceled.
      */
     void canceled();
+
+    /**
+     * Callback operation to update the percentage progress of the firmware update.
+     * This method can be used to provide detailed progress information additional to the sequence or even without a
+     * previous defined sequence.
+     * 
+     * @param progress the progress between 0 and 100
+     * 
+     * @throws IllegalArgumentException if given progress is < 0 or > 100
+     * @throws IllegalArgumentException if given progress is smaller than old progress
```

Comment:
pls check also the method signature for the other operations... most of them throw an IllegalStateException (and an IllegalArgumentException) that is not described